include/aspect/heating_model/latent_heat.h

```diff
@@ -0,0 +1,86 @@
+/*
+  Copyright (C) 2014 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__heating_model_latent_heat_h
+#define __aspect__heating_model_latent_heat_h
+
+#include <aspect/heating_model/interface.h>
+
+namespace aspect
+{
+  namespace HeatingModel
+  {
+    using namespace dealii;
+
+    /**
+     * A class that implements a standard formulation of latent heat.
+     * This includes a left hand side and a right hand side term:
+     *
+     * The left hand side is
+     *   $ -\rho T frac{\partial S}{\partial T} \frac{D T}{D t}$
+     * and is added to the rho cp term.
+     *
+     * The right-hand side term from latent heating is
+     *   $\frac{\partial S}{\partial p} T \rho (v \dot \nabla p)$.
+     *
+     * Formulation modified after Christensen & Yuen, 1985.
```

Comment:
Let's not worry about it here then. Just say something like "where T, v, and p are the solutions from the previous time step or are extrapolated from there." @rrgrove6 has plans to document issues such as this somewhere in the manual.